packages/analytics/shippers/fullstory/src/fullstory_shipper.ts

```diff
@@ -77,9 +91,56 @@ export class FullStoryShipper implements IShipper {
     config: FullStoryShipperConfig,
     private readonly initContext: AnalyticsClientInitContext
   ) {
-    const { eventTypesAllowlist, ...snippetConfig } = config;
+    const { eventTypesAllowlist, pageVarsDebounceTimeMs = 500, ...snippetConfig } = config;
     this.fullStoryApi = loadSnippet(snippetConfig);
     this.eventTypesAllowlist = eventTypesAllowlist;
+
+    this.subscriptions.add(
+      this.userContext$
+        .pipe(
+          distinct(({ userId, isElasticCloudUser, cloudIsElasticStaffOwned, cloudTrialEndDate }) =>
+            [userId, isElasticCloudUser, cloudIsElasticStaffOwned, cloudTrialEndDate].join('-')
+          )
+        )
+        .subscribe((userVars) => this.updateUserVars(userVars))
+    );
+
+    this.subscriptions.add(
+      this.pageContext$
+        .pipe(
+          map((newContext) => {
+            // Cherry-picking fields because FS limits the number of fields that can be sent.
+            // > Note: You can capture up to 20 unique page properties (exclusive of pageName) for any given page
+            // > and up to 500 unique page properties across all pages.
+            // https://help.fullstory.com/hc/en-us/articles/1500004101581-FS-setVars-API-Sending-custom-page-data-to-FullStory
+            return PAGE_VARS_KEYS.reduce((acc, key) => {
+              if (has(newContext, key)) {
+                set(acc, key, get(newContext, key));
+              }
+              return acc;
+            }, {} as Partial<FullStoryPageContext> & Record<string, unknown>);
+          }),
+          filter((pageVars) => Object.keys(pageVars).length > 0),
+          // Wait for anything to actually change.
+          distinct((pageVars) => {
+            const sortedKeys = Object.keys(pageVars).sort();
+            return sortedKeys.map((key) => pageVars[key]).join('-');
+          }),
```

Comment:
FWIW, I think this is more useful than the actual debounceTime. My gut feeling after looking at the traces in FS is that the "context" update was triggered by [the page URL change](https://github.com/elastic/kibana/blob/97dc2ecba127e997e5229b499d83a0a38072c043/packages/core/application/core-application-browser-internal/src/register_analytics_context_provider.ts#L19-L25), but we don't ship those properties to FS (since it already captures it on its own).